addons/binding/org.openhab.binding.systeminfo/src/main/java/org/openhab/binding/systeminfo/model/OshiSysteminfo.java

```diff
@@ -393,6 +427,20 @@ public DecimalType getSwapAvailablePercent() {
         }
     }
 
+    @Override
+    public DecimalType getSwapUsedPercent() {
+        long usedSwap = memory.getSwapUsed();
+        long totalSwap = memory.getSwapTotal();
+        BigDecimal usedPercent;
```

Comment:
A reason why you separated this declaration of the initialization?

It could make sense to have only the fetching of the `totalSwap` outside the `if`?